cli/internal/helm/client.go

```diff
@@ -153,70 +173,54 @@ func (c *Client) currentVersion(release string) (string, error) {
 var ErrConfirmationMissing = errors.New("action requires user confirmation")
 
 func (c *Client) upgradeRelease(
-	ctx context.Context, timeout time.Duration, conf *config.Config, chartPath string, releaseName string, hasCRDs bool, allowDestructive bool,
+	ctx context.Context, timeout time.Duration, conf *config.Config, chart *chart.Chart, allowDestructive bool,
 ) error {
-	chart, err := loadChartsDir(helmFS, chartPath)
-	if err != nil {
-		return fmt.Errorf("loading chart: %w", err)
-	}
-
 	// We need to load all values that can be statically loaded before merging them with the cluster
 	// values. Otherwise the templates are not rendered correctly.
 	k8sVersion, err := versions.NewValidK8sVersion(conf.KubernetesVersion)
 	if err != nil {
 		return fmt.Errorf("invalid k8s version: %w", err)
 	}
 	loader := NewLoader(conf.GetProvider(), k8sVersion)
+
 	var values map[string]any
-	switch releaseName {
-	case ciliumReleaseName:
+	var info chartInfo
+
+	switch chart.Metadata.Name {
+	case ciliumInfo.chartName:
+		info = ciliumInfo
 		values, err = loader.loadCiliumValues()
```

Comment:
I applied your suggestions but kept the `chartInfo` type for now. Reason: I am hoping to unify the `loadCilium`, `loadCertManager`, `loadOperators`, `loadConstellationServices` in loader.go into a single function, hopefully making the code more concise & understandable. But I would do that in a separate PR to keep PRs small and simple.

If you think this is a bad idea, please tell me. If it turns out to not work as I expect I would remove the chartInfo type again.

Reply:
Only allowed for some minor change: https://github.com/edgelesssys/constellation/pull/1441
Not sure if it's better that way. "Feels" better because there is slightly less code duplication. I guess the DRY principle sits deep in me :D

But also doesn't make it worse. So :man_shrugging: ..